dotnet-authserver/src/TeacherIdentity.AuthServer/Controllers/AuthorizationController.cs

```diff
@@ -126,13 +113,23 @@ public async Task<IActionResult> Authorize()
                 userRequirements,
                 GetCallbackUrl(journeyId),
                 startedAt: _clock.UtcNow,
-                authenticateResult.Succeeded != true,
-                authStateInitData,
                 sessionId,
                 oAuthState: new OAuthAuthorizationState(request.ClientId!, request.Scope!, request.RedirectUri)
                 {
                     TrnRequirementType = trnRequirementType
-                });
+                },
+                authenticateResult.Succeeded != true);
+
+            var signedInUser = await GetSignedInUser(authenticateResult, userRequirements);
+
+            if (trnToken is not null)
+            {
+                await _trnTokenHelper.InitializeAuthenticationStateWithToken(signedInUser, authenticationState, trnToken, HttpContext);
```

Comment:
What is happening here:
If there is no signed in user, but we find an existing user with an email match, we add the existing users details to auth state. When we hit lines 189, 190
` var signInJourney = _signInJourneyProvider.GetSignInJourney(authenticationState, HttpContext);
            return Redirect(signInJourney.GetStartStepUrl());`
We get redirected to this endpoint again, except this time we DO have a signed in user. I don't know why 
`return Redirect(signInJourney.GetStartStepUrl())` ends up back here, so this is the best solution I've come up with so far